File: retort-core/src/main/java/co/kaioru/retort/annotation/IAnnotationGenerator.java
package co.kaioru.retort.annotation;

import co.kaioru.retort.ICommand;
import co.kaioru.retort.ICommandContext;

import java.lang.reflect.Method;

public interface IAnnotationGenerator<I extends ICommandContext, O> extends IAnnotationAdapter<I, O> {

    ICommand<I, O> generate(Object object, Method method);

}

File: retort-annotations/src/main/kotlin/co/kaioru/retort/annotation/IAnnotationGenerator.kt
package co.kaioru.retort.annotation

import co.kaioru.retort.ICommand
import co.kaioru.retort.annotation.exceptions.AnnotationAdapterException
import java.lang.reflect.Method

interface IAnnotationGenerator<I, O> : IAnnotationAdapter<I, O> {
    @Throws(AnnotationAdapterException::class)
    fun generate(any: Any, method: Method): ICommand<I, O>
}